### PR TITLE
Add more functionality to ISC-DHCP configuration

### PR DESCRIPTION
**Pull request by pprindeville**
Maintainer: me
Compile tested: x86_64, generic, HEAD (759ee14)
Run tested: built and installed .ipk on production router

Description:

CentOS/Fedora/RHEL all need the domain as one of the DHCP options they are given or bad things happen when using DHCP and NetworkManager.

Adding `ntp-servers` and `domain-search` as low-hanging fruit as well.

### CODE REVIEW

**Comment by pprindeville** [comment]
Yeah, I thought about that... okay, let me circle back...

**Comment by pprindeville** [comment]
Btw, the argument, if if matches `\d+` or `\d+\.\d+\.\d+\.\d+`, `true`, or `false`, wouldn't need quotes, but anything else would, right?  What's the best way to do matching in our braindead shell?

**Comment by pprindeville** [comment]
> As dnsmasq apparently doesn't seem to support adding additional search-domains handed out via DHCP I'm ok with this being in the isc_dhcpd section.

It could, but you'd have to do it the same way... as raw DHCP options.

**Comment by pprindeville** [comment]
Hmm...  We don't have arrays in this version of shell... but the value argument for `dhcp_option` can contain spaces.  How to handle that?

**Comment by pprindeville** [comment]
```
        list dhcp_option 'option:ntp-server, 192.168.0.4'
```

Isn't the `option:` prefix redundant?

**Comment by dangowrt** [comment]
It's the existing syntax which is inherited from dnsmasq...

**Comment by pprindeville** [comment]
> It's the existing syntax which is inherited from dnsmasq...

Well, I've made the `option:` prefix optional.

**Comment by pprindeville** [comment]
If we have:

```
list dhcp_option 'option:ntp-servers, 192.168.1.40'
list dhcp_option 'option:ip-forwarding false'
list dhcp_option 'option:max-dgram-reassembly 32768'
list dhcp_option 'option:domain-search,redfish-solutions.com,redfish-solutions.net'
```

This _should_ generate:

```
 option ntp-servers 192.168.1.40;
 option ip-forwarding false;
 option max-dgram-reassembly 32768;
 option domain-search "redfish-solutions.com", "redfish-solutions.net";
```

because booleans, numbers, and IP addresses don't require quoting...  everything else (i.e. text) does.

**Comment by pprindeville** [comment]
> Guess I'll need to write an awk script which takes a parameter and determines its type.

Done.  Please re-review.

File: net/isc-dhcp/files/dhcpd.init [hunk 1]
```diff
@@ -120,7 +120,44 @@ static_hosts() {
...
+
+	local tag="${tuple%%,*}"
+	local values="${tuple#$tag,}"
+
+	local formatted value
+	local IFS=$', \n'
+	for value in $values; do
+		# detect type of $value and quote if necessary
+		case $(typeof "$value") in
+		ip|bool|integer|string)
```
Comment (nmeyerhans) [inline net/isc-dhcp/files/dhcpd.init]: Strings don't need to be quoted?

File: net/isc-dhcp/files/dhcpd.init [hunk 2]
```diff
@@ -120,7 +120,44 @@ static_hosts() {
 	config_foreach static_host_add host "$@"
 }
 
+typeof() {
+	echo "$1" | awk '
+/^\d+\.\d+\.\d+\.d+$/		{ print "ip\n"; next; }
+/^(true|false)$/		{ print "bool\n"; next; }
+/^\d+$/				{ print "integer\n"; next; }
```
Comment (nmeyerhans) [inline net/isc-dhcp/files/dhcpd.init]: indentation looks off here.

Comment (pprindeville) [reply net/isc-dhcp/files/dhcpd.init]: That's the '+' throwing it off.

File: net/isc-dhcp/files/dhcpd.init (hunk 1 above)
Comment (pprindeville) [reply net/isc-dhcp/files/dhcpd.init]: They already are.  See line 128.

**Comment by pprindeville** [comment]
@dangowrt How does it look now?

**Comment by pprindeville** [comment]
Tried to add @EricLuehrsen to reviewers..

**Comment by pprindeville** [comment]
Not heard back from @dangowrt, proceeding anyway.

**Pull request merged**